tensorbay/opendataset/UAVDT/loader.py

```diff
@@ -0,0 +1,175 @@
+
+    The file structure of UAVDT looks like:
+
+        <path>
+            M_attr/
+                test/
+                    M0203_attr.txt
+                    ...
+                train/
+                    M0101_attr.txt
+                    ...
+            UAVDT_Benchmark_M/
+                M0101/
+                    img000001.jpg
+                    ...
+                ...
+            UAV-benchmark-MOTD_v1.0/
+                GT/
+                    M0101_gt_ignore.txt
+                    M0101_gt.txt
+                    M0101_gt_whole.txt
+                    ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = Dataset(_DATASET_NAME)
+    dataset.notes.is_continuous = True
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for mode in _MODES:
+        for sequence_attributes_path in glob(os.path.join(root_path, "M_attr", mode, "*.txt")):
+            sequence = os.path.basename(sequence_attributes_path)[:5]
+            segment = dataset.create_segment(f"{mode}-{sequence}")
+
+            sequence_attributes = _extract_attr_txt_to_dict(
+                sequence_attributes_path, dataset.catalog.classification.attributes.keys()
+            )
+            classification = Classification(attributes=sequence_attributes)
+
+            frame_id_ground_truth_map = _extract_gt_txt_to_dict(
+                root_path,
+                sequence,
+                dataset.catalog.box2d,
+            )
+
+            image_paths = glob(os.path.join(root_path, "UAV-benchmark-M", sequence, "*.jpg"))
+            for image_path in image_paths:
+                data = Data(image_path)
+                # The image_name looks like `img<frame_id>.jpg`.
+                # The frame_id is consist of six digital numbers.
+                ground_truths = frame_id_ground_truth_map[int(os.path.basename(image_path)[3:9])]
+                data.label.classification = classification
+                data.label.box2d = _convert_ground_truths_to_labels(ground_truths)
+                segment.append(data)
+
+    return dataset
+
+
+def _extract_attr_txt_to_dict(
+    sequence_attributes_path: str, attribute_names: Tuple[str, ...]
+) -> Dict[str, bool]:
+    with open(sequence_attributes_path) as fp:
+        csv_reader = csv.reader(fp)
+        elements = next(csv_reader)
+        attributes = {
+            attribute_name: bool(int(value))
+            for attribute_name, value in zip(attribute_names, elements)
+        }
+
+    return attributes
+
+
+def _extract_gt_txt_to_dict(
+    path: str,
+    sequence: str,
+    box2d_catalog: Box2DSubcatalog,
+) -> Dict[int, List[Dict[str, Any]]]:
+    attributes = box2d_catalog.attributes
+    catagories = box2d_catalog.categories.keys()
```

Comment:
it is weird to name `categories.keys()` as `categories` but `attributes` to `attributes`